fix: Parameters with default values are optional

bfanger opened the pull request:
I've marked the properties of parameters that have default values as optional (`x: number` to `x?: number`)

The current situation marks the properties of the interface as required, which cause errors when properties are omitted:
<img width="850" alt="screenshot" src="https://user-images.githubusercontent.com/207248/85951592-527ccc00-b964-11ea-9064-fc67687cde7c.png">
> Argument of type '{ x: number; }' is not assignable to parameter of type 'FlyParams'.
  Type '{ x: number; }' is missing the following properties from type 'FlyParams': delay, duration, easing, y, opacity

dkzlv commented:
@bfanger As far as I understand your solution would still require the developer to pass an empty object to all these animation functions like this: `transition:fade={{}}`.

In order to make it completely compatible to the real interface we need to make the argument itself an optional one.

bfanger commented:
@dkzlv `<div transition:fly>` already worked without errors, I first thought this was a feature/hack of the language-server and changed the second parameter to optional, but i've reverted that change. As it turns out, when you don't provide a value for a transition svelte will call the function with an empty object as 2nd parameter. 

`<div transition:fly>` is shorthand for `<div transition:fly={{}}>`

Conduitry commented:
The fixed types have been published in 3.24.0.
